src/builder.rs

```diff
@@ -788,13 +806,21 @@ impl<S: InProgressSignatures, V> Bundle<InProgress<Unproven, S>, V> {
     ) -> Bundle<InProgress<Proven, S>, V> {
         let total_progress =
             self.shielded_spends().len() as u32 + self.shielded_outputs().len() as u32;
-        self.map_authorization(CreateProofs::new(
+        let mut cp = CreateProofs::new(
```

Comment:
I was unable to find a way to pass a _mutable_ context through when constructing the `MapAuth` in terms of a tuple of functions. When I saw how Orchard did it, it seemed much simpler, so I went with that.

Since `TryMapAuth` is unused, we might as well get rid of it.